server/audit_math/suite.py

```diff
@@ -1,718 +0,0 @@
-
-        """
-        Compute a p-value for a winner-loser pair for this strata based on its math type.
-
-        Inputs:
-            reported_margin: the alternative hypothesis margin, in ballots
-            winner, loser: the winner-loser pair to evaluate the hypothesis on
-            null_lambda: the Fisher's combining lambda for the null hypothesis.
-
-        Outputs:
-            pvalue - the pvalue for the hypothesis given the null_lambda
-        """
-
-        if self.sample_size == 0 or reported_margin == 0:
-            return 1.0
-
-        if self.sample_size == self.num_ballots:
-            return 0.0
-
-        o1, o2, u1, u2 = (
-            self.misstatements[(winner, loser)]["o1"],
-            self.misstatements[(winner, loser)]["o2"],
-            self.misstatements[(winner, loser)]["u1"],
-            self.misstatements[(winner, loser)]["u2"],
-        )
-
-        U_s = Decimal(2 * self.num_ballots / reported_margin)
-        gamma = Decimal(GAMMA)
-        multiplier = 1 - Decimal(null_lambda) / (gamma * U_s)
-
-        # This represents an invalid alternative, because lambda is too big.
-        if multiplier <= 0:
-            raise ValueError("Alternative hypothesis is invalid!")
-
-        log_pvalue = (
-            self.sample_size * multiplier.ln()
-            - o1 * (1 - 1 / (2 * gamma)).ln()
-            - o2 * (1 - 1 / gamma).ln()
-            - u1 * (1 + 1 / (2 * gamma)).ln()
-            - u2 * (1 + 1 / gamma).ln()
-        )
-        pvalue = (log_pvalue).exp()
-        return float(np.min([float(pvalue), 1.0]))  # cast for the typechecker
-
-
-def maximize_fisher_combined_pvalue(
-    alpha: float,
-    contest: Contest,
-    bp_stratum: BallotPollingStratum,
-    cvr_stratum: BallotComparisonStratum,
-    winner: str,
-    loser: str,
-    stepsize: float = 0.05,
-) -> float:
-    """
-    Grid search to find the maximum P-value.
-    Find the smallest Fisher's combined statistic for P-values obtained
-    by testing two null hypotheses at level alpha using data X=(X1, X2).
-
-    This was largely taken from CORLA18.
-
-    Inputs:
-        alpha: the risk limit as a fraction
-        contest: the contest being audited
-        bp_stratum: a stratum object containing the ballot polling stratum
-        cvr_stratum: a stratum object containing the ballot comparison stratum
-        winner, loser: the winner-loser pair to maximuze the p-value over
-
-    Outputs:
-        a maximum p-value for the Fisher's combining function given the two
-        strata.
-    """
-    maximized_pvalue = 0.0
-    # find range of possible lambda
-    cvr_winner_votes = cvr_stratum.vote_totals[winner]
-    bp_winner_votes = bp_stratum.vote_totals[winner]
-
-    cvr_loser_votes = cvr_stratum.vote_totals[loser]
-    bp_loser_votes = bp_stratum.vote_totals[loser]
-
-    V = cvr_winner_votes - cvr_loser_votes + bp_winner_votes - bp_loser_votes
```

Comment:
Is `V` equivalent to `reported_margin` below?  I believe it is.

If yes, then I think you can remove `vote_totals` from the ballot comparison stratum entirely, since they aren't used anywhere else.

Reply:
I think that's correct.

Reply:
Ah, no we do need cvr_winner/loser votes below.